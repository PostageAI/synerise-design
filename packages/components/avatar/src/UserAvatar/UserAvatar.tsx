import * as React from 'react';
import { UserM, UserS } from '@synerise/ds-icon/dist/icons';
import theme from '@synerise/ds-core/dist/js/DSProvider/ThemeProvider/theme';
import Icon from '@synerise/ds-icon';
import Badge from '@synerise/ds-badge';

import Avatar from '../Avatar';
import { getUserText ,addIconColor } from '../utils';
import { UserAvatarProps } from '../Avatar.types';

export const DEFAULT_COLOR = 'grey';
export const DEFAULT_COLOR_HUE = '500';

const UserAvatar: React.FC<UserAvatarProps> = ({ 
  backgroundColor = 'grey', 
  size = 'small', 
  text,
  tooltip,
  badgeStatus,
  firstName = '',
  lastName = '',
  email,
  src,
  iconComponent,
  children,
  disabled,
  ...restProps
}) => {
  const avatarText = getUserText(firstName, lastName, src, text);
  const defaultTooltip = { title: `${firstName} ${lastName}`.trim(), description: email || '' };
  const avatarTooltip = tooltip === undefined && (defaultTooltip.title || defaultTooltip.description) ? defaultTooltip : tooltip;

  const iconColor = theme.palette[`${DEFAULT_COLOR}-${DEFAULT_COLOR_HUE}`];
  const iconElement = addIconColor(iconComponent, iconColor);

  const icon = !avatarText 
    ? iconElement || <Icon component={size === 'small' ? <UserS/> : <UserM />} color={theme.palette['grey-500']} /> 
    : null;

  const avatar = (
    <Avatar
      iconComponent={icon}
      shape="circle"
      hasStatus={disabled === true ? false : !!badgeStatus}
      backgroundColor={avatarText ? backgroundColor : 'grey'}
      backgroundColorHue={avatarText ? '500' : '100'}
      size={size}
      src={src}
      tooltip={avatarTooltip}
      disabled={disabled}
      {...restProps}
    >
      {children || avatarText}
    </Avatar>
  );
  
  return badgeStatus ? <Badge status={badgeStatus}>{avatar}</Badge> : avatar;
};

export default UserAvatar;
