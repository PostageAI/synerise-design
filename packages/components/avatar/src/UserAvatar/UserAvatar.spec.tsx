import * as React from 'react';
import { renderWithProvider } from '@synerise/ds-utils/dist/testing';
import UserAvatar from './UserAvatar';

const testUser = {
  firstName: 'John',
  lastName: 'Doe',
  email: 'john.doe@synerise.com',
};

const testUserImage = {
  firstName: 'John',
  lastName: 'Doe',
  email: 'john.doe@synerise.com',
  avatar: 'http://image.com/image.jpg',
};

describe('UserAvatar', () => {
  test('should match snapshot', () => {
    const userAvatar = renderWithProvider(<UserAvatar />);
    expect(userAvatar).toMatchSnapshot();
  });
  test('should match snapshot with user data', () => {
    const userAvatar = renderWithProvider(<UserAvatar user={testUser} />);
    expect(userAvatar).toMatchSnapshot();
  });
  test('should match snapshot with user data and avatar', () => {
    const userAvatar = renderWithProvider(<UserAvatar user={testUserImage} />);
    expect(userAvatar).toMatchSnapshot();
  });
});
