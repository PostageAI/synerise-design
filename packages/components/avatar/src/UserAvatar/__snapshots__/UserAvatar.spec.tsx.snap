// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="ant-avatar Avatarstyles-xcn5o4-1 hiogYD ds-avatar ant-avatar-circle"
      >
        <span
          class="ant-avatar-string"
          style="transform: scale(1) translateX(-50%);"
        >
          <div
            class="Iconstyles__IconContainer-aqhvae-0 beTcfQ ds-icon "
            color="#949ea6"
            size="24"
          >
            <svg
              class="user-m"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 11.7a3.874 3.874 0 10-3.869-3.874A3.876 3.876 0 0012 11.7zm0-6.248a2.374 2.374 0 11-2.369 2.374A2.374 2.374 0 0112 5.452zM15.225 12.75h-6.45a4.755 4.755 0 00-4.75 4.75v.76a1.752 1.752 0 001.75 1.75h12.45a1.752 1.752 0 001.75-1.75v-.76a4.755 4.755 0 00-4.75-4.75zm3.25 5.51a.25.25 0 01-.25.25H5.775a.25.25 0 01-.25-.25v-.76a3.254 3.254 0 013.25-3.25h6.45a3.254 3.254 0 013.25 3.25z"
              />
            </svg>
          </div>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="ant-avatar Avatarstyles-xcn5o4-1 hiogYD ds-avatar ant-avatar-circle"
    >
      <span
        class="ant-avatar-string"
        style="transform: scale(1) translateX(-50%);"
      >
        <div
          class="Iconstyles__IconContainer-aqhvae-0 beTcfQ ds-icon "
          color="#949ea6"
          size="24"
        >
          <svg
            class="user-m"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 11.7a3.874 3.874 0 10-3.869-3.874A3.876 3.876 0 0012 11.7zm0-6.248a2.374 2.374 0 11-2.369 2.374A2.374 2.374 0 0112 5.452zM15.225 12.75h-6.45a4.755 4.755 0 00-4.75 4.75v.76a1.752 1.752 0 001.75 1.75h12.45a1.752 1.752 0 001.75-1.75v-.76a4.755 4.755 0 00-4.75-4.75zm3.25 5.51a.25.25 0 01-.25.25H5.775a.25.25 0 01-.25-.25v-.76a3.254 3.254 0 013.25-3.25h6.45a3.254 3.254 0 013.25 3.25z"
            />
          </svg>
        </div>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserAvatar should match snapshot with user data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="ant-avatar Avatarstyles-xcn5o4-1 eUarBN ds-avatar ant-avatar-circle"
      >
        <span
          class="ant-avatar-string"
          style="transform: scale(1) translateX(-50%);"
        >
          JD
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="ant-avatar Avatarstyles-xcn5o4-1 eUarBN ds-avatar ant-avatar-circle"
    >
      <span
        class="ant-avatar-string"
        style="transform: scale(1) translateX(-50%);"
      >
        JD
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
